x/mongo/driver/description/topology.go

```diff
@@ -41,39 +38,24 @@ type TopologyDiff struct {
 func DiffTopology(old, new Topology) TopologyDiff {
 	var diff TopologyDiff
 
-	// TODO: do this without sorting...
-	oldServers := serverSorter(old.Servers)
-	newServers := serverSorter(new.Servers)
-
-	sort.Sort(oldServers)
-	sort.Sort(newServers)
-
-	i := 0
-	j := 0
-	for {
-		if i < len(oldServers) && j < len(newServers) {
-			comp := strings.Compare(oldServers[i].Addr.String(), newServers[j].Addr.String())
-			switch comp {
-			case 1:
-				//left is bigger than
-				diff.Added = append(diff.Added, newServers[j])
-				j++
-			case -1:
-				// right is bigger
-				diff.Removed = append(diff.Removed, oldServers[i])
-				i++
-			case 0:
-				i++
-				j++
-			}
-		} else if i < len(oldServers) {
-			diff.Removed = append(diff.Removed, oldServers[i])
-			i++
-		} else if j < len(newServers) {
-			diff.Added = append(diff.Added, newServers[j])
-			j++
+	oldServers := make(map[string]bool)
+	for _, s := range old.Servers {
+		oldServers[s.Addr.String()] = true
+	}
+
+	for _, s := range new.Servers {
+		addr := s.Addr.String()
+		if oldServers[addr] {
+			delete(oldServers, addr)
 		} else {
-			break
+			diff.Added = append(diff.Added, s)
+		}
+	}
+
+	for _, s := range old.Servers {
```

Comment:
Is there a reason this iterates over `old.Servers` rather than `oldServers` like `DiffHostList` does?

Reply:
`oldServers` only gives you the string address, whereas we need the full `Server` (`DiffHostList` operates on addresses which is why we can do that simplification there).